dumpling/export/sql.go

```diff
@@ -827,13 +827,16 @@ func getSnapshot(db *sql.Conn) (string, error) {
 	return str[snapshotFieldIndex], nil
 }
 
-func isUnknownSystemVariableErr(err error) bool {
-	return strings.Contains(err.Error(), "Unknown system variable")
+func isUnknownSystemVariableErr(err error, ignorableSnapshot bool) bool {
+	errStr := err.Error()
+	return strings.Contains(err.Error(), "Unknown system variable") ||
```

Comment:
Another possible solution is to ignore all set session tidb_snapshot error when ignorableSnapshot is true. Because I'm not so sure whether tidb's error will change in the future or not.